tests/activity_scheduling_helper.cpp

```diff
@@ -103,14 +103,14 @@ weariness_events do_activity( tasklist tasks )
         // How many turn's we've been at it
         time_duration turns = 0_seconds;
         while( turns <= task.interval && !task.instantaneous() ) {
-            // Start each turn with a fresh set of moves
-            guy.moves = 100;
-            task.do_turn( guy );
             // Advance a turn
             calendar::turn += 1_turns;
             turns += 1_seconds;
             // Consume food, become weary, etc
             guy.update_body();
+            // Start each turn with a fresh set of moves
+            guy.moves = 100;
+            task.do_turn( guy );
         }
```

Comment:
I can see why putting the do_turn after advancing turns makes sense. But why update body prior to do_turn?

Reply:
Because that's the order in which it occurs in game::do_turn().
Technically moves are incremented at the end of the turn instead of the start, but update_body() shouldn't be interacting with moves, so it shouldn't matter.